File: june/lt4.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    for(cin>>t;t--;)
    {
        int n,k,x=0;
        cin>>n>>k;
        vector<int> bits(32,0);
        for(int i=0;i<n;i++)
        {
            int input,j=0;
            cin>>input;
            do
            {
                bits[j++]+=(input&1);
            } while (input>>=1);
        }

        int ans=0;

        while(k--)
        {
            int c=0,p=-1;
            for(int i=0;i<32;i++) if(bits[i] && bits[i]*pow(2,i)>c) c=bits[i]*pow(2,i),p=i;
            if(p==-1) break;
            else
            {
                bits[p]=0;
                ans+=pow(2,p);
            }
        }
        cout<<ans<<endl;
    }
}

File: june/lt4.py
for _ in range(int(input())):
    n,k = map(int,input().split())
    ll = list(map(int,input().split()))
    xx = [0]*32
    for i in ll:
        x = bin(i)[2:]
        for j in range(len(x)):
            if x[j]=='1':xx[j]+=1
    ans=0
    while k>0:
        k-=1;c=0;p=-1
        for i in range(32):
            if(xx[i]>0):
                if(ans+xx[i]*(2**i)>ans+c): c=xx[i]*2**i;p=2**i
        if(p==-1): break
        else: ans+=p;xx[i]=0
    print(ans)